recipes/sleef/all/conanfile.py

```diff
@@ -24,75 +28,79 @@ class SleefConan(ConanFile):
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def validate(self):
         if self.settings.os == "Windows" and self.options.shared:
-            raise ConanInvalidConfiguration("shared sleef not supported on Windows, it produces runtime errors")
+            raise ConanInvalidConfiguration(
+                "shared sleef not supported on Windows, it produces runtime errors"
+            )
+        if self.settings.compiler == "apple-clang":
+            if cross_building(self):
+                # Fails with "No rule to make target `/bin/mkrename'"
+                # https://github.com/shibatch/sleef/issues/308
+                raise ConanInvalidConfiguration(f"{self.ref} does not support cross-building with apple-clang")
+            if self.settings.arch == "armv8":
+                # clang: error: the clang compiler does not support '-march=armv7-a'
+                # clang: warning: argument unused during compilation: '-mfpu=vfpv4' [-Wunused-command-line-argument]
+                # clang: warning: argument unused during compilation: '-arch arm64' [-Wunused-command-line-argument]
+                # clang: warning: argument unused during compilation: '-mmacosx-version-min=11.0' [-Wunused-command-line-argument]
+                raise ConanInvalidConfiguration(f"armv8 is not supported for {self.ref}")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename(self.name + "-" + self.version, self._source_subfolder)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["BUILD_STATIC_TEST_BINS"] = False
-        self._cmake.definitions["ENABLE_LTO"] = False
-        self._cmake.definitions["BUILD_LIBM"] = True
-        self._cmake.definitions["BUILD_DFT"] = False
-        self._cmake.definitions["BUILD_QUAD"] = False
-        self._cmake.definitions["BUILD_GNUABI_LIBS"] = False
-        self._cmake.definitions["BUILD_TESTS"] = False
-        self._cmake.definitions["BUILD_INLINE_HEADERS"] = False
-        self._cmake.definitions["SLEEF_TEST_ALL_IUT"] = False
-        self._cmake.definitions["SLEEF_SHOW_CONFIG"] = True
-        self._cmake.definitions["SLEEF_SHOW_ERROR_LOG"] = False
-        self._cmake.definitions["ENFORCE_TESTER"] = False
-        self._cmake.definitions["ENFORCE_TESTER3"] = False
-        self._cmake.definitions["ENABLE_ALTDIV"] = False
-        self._cmake.definitions["ENABLE_ALTSQRT"] = False
-        self._cmake.definitions["DISABLE_FFTW"] = True
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
```

Comment:
Some options changed in 3.6, including for testing. Please, check in the new release.